test/end2end_test.go

```diff
@@ -6954,12 +6958,19 @@ func (s) TestGoAwayThenClose(t *testing.T) {
 
 	client := testpb.NewTestServiceClient(cc)
 
-	// Should go on connection 1. We use a long-lived RPC because it will cause GracefulStop to send GO_AWAY, but the
-	// connection doesn't get closed until the server stops and the client receives.
+	// We make a streaming RPC and do an one-message-round-trip to make sure
+	// it's created on connection 1.
+	//
+	// We use a long-lived RPC because it will cause GracefulStop to send
+	// GO_AWAY, but the connection doesn't get closed until the server stops and
+	// the client receives the error.
 	stream, err := client.FullDuplexCall(ctx)
 	if err != nil {
 		t.Fatalf("FullDuplexCall(_) = _, %v; want _, nil", err)
 	}
+	if _, err = stream.Recv(); err != nil {
+		t.Fatalf("unexpected error from first recv: %v", err)
+	}
```

Comment:
Somewhat related... Do we have another test that will do sends/receives/finish RPCs on a gracefully-stopped-but-not-fully-stopped server?  If not, then adding another `Recv` after `conn2Established` but before `s1.Stop()` would be a cheap way of testing it.

Reply:
That's covered in the other GoAway tests: https://github.com/grpc/grpc-go/blob/master/test/end2end_test.go#L1114